source/MaterialXTest/ShaderValid.cpp

```diff
@@ -281,29 +313,155 @@ TEST_CASE("GLSL Validation from HwShader", "[shadervalid]")
         REQUIRE(hwShader->getSourceCode(mx::HwShader::PIXEL_STAGE).length() > 0);
         REQUIRE(hwShader->getSourceCode(mx::HwShader::VERTEX_STAGE).length() > 0);
 
+        std::ofstream file;
+        file.open(nodePtr->getName() + ".vert");
+        file << shader->getSourceCode(mx::HwShader::VERTEX_STAGE);
+        file.close();
+        file.open(nodePtr->getName() + ".frag");
+        file << shader->getSourceCode(mx::HwShader::PIXEL_STAGE);
+        file.close();
+
+
+        validator.setStages(hwShader);
+        unsigned int programId = validator.createProgram();
+        REQUIRE(programId > 0);
+        const mx::GlslValidator::ProgramInputMap& uniforms = validator.getUniformsList();
+        for (auto input : uniforms)
+        {
+            unsigned int gltype = input.second->_gltype;
+            int location = input.second->_location;
+            int size = input.second->_size;
+            std::string type = input.second->_typeString;
+            std::string value = input.second->_value ? input.second->_value->getValueString() : "<none>";
+            std::cout << "Program Uniform: \"" << input.first
+                << "\". Location=" << location
+                << ". Type=" << std::hex << gltype
+                << ". Size=" << std::dec << size
+                << ". TypeString=" << type
+                << ". Value=" << value << "."
+                << std::endl;
+
+        }
+        const mx::GlslValidator::ProgramInputMap& attributes = validator.getAttributesList();
+        for (auto input : attributes)
+        {
+            unsigned int gltype = input.second->_gltype;
+            int location = input.second->_location;
+            int size = input.second->_size;
+            std::string type = input.second->_typeString;
+            std::string value = input.second->_value ? input.second->_value->getValueString() : "<none>";
+            std::cout << "Program Attribute: \"" << input.first
+                << "\". Location=" << location
+                << ". Type=" << std::hex << gltype
+                << ". Size=" << std::dec << size
+                << ". TypeString=" << type 
+                << ". Value=" << value << "."
+                << std::endl;
+        }
+        validator.render();
+        std::string fileName = nodePtr->getName() + ".exr";
+        validator.save(fileName);
+    }
+#if 0
```

Comment:
Niklas, I will turn this on afterwards. Your change for the doc specified here use unknown nodedefs for now so turning off temporarily.

Reply:
Oh yes, it's the layeredbsdf node that is now named mixbsdf (to be consistent with mixedf, mixvdf, etc.) I can fix this and enabled the tests again when you have merged.

Reply:
But if it's only an example surface shader you need to test with here you could also load the SubGraphs.mtlx example file and use the second example graph there which is a surface shader graph.

We could add in a few more examples files to the Examples folder, and use for testing, so we don't have to create the graphs from scratch each time. (but that's a separate task)

Reply:
Okay. I'll look into this afterwards then.